engine/Core.php

```diff
@@ -8,14 +8,22 @@ class Core {
 
     public function __construct() {
 
-        $pos = strpos(ASSUMED_URL, MODULE_ASSETS_TRIGGER);
+        if (strpos(ASSUMED_URL, '/vendor/')) {
+            $this->serve_vendor_asset();
+        } elseif(strpos(ASSUMED_URL, MODULE_ASSETS_TRIGGER) === false) {
+            $this->serve_controller();
+        } else {
+            $this->serve_module_asset();
+        }
 
-        if (strpos(ASSUMED_URL, 'vendor/')) {     
+    }
 
-            $vendor_file_path = explode('vendor/', ASSUMED_URL)[1];
-            $vendor_file_path = '../vendor/'.$vendor_file_path;
+    private function serve_vendor_asset() {
+        $vendor_file_path = explode('/vendor/', ASSUMED_URL)[1];
+        $vendor_file_path = '../vendor/'.$vendor_file_path;
+        if (file_exists($vendor_file_path)) {
             $content_type = mime_content_type($vendor_file_path);
-
+            die($vendor_file_path);
```

Comment:
you forgot to remove this?

Reply:
Yes! Thank you, I didnt notice I left it in there after testing.

Reply:
I must confess, I don't entirely understand the pull request.  

Please don't assume that I'm being smug or critical when I say this.  I'm not.  My head is elsewhere just now and I'm having a hard time wrapping my mind around this one.  Reach out to me if you'd like to discuss the pull request further.  I very much appreciate you taking the time to look at the code and submit your pull request.  Thank you!  And thanks to Derek, too!